### DISCUSSION TITLE
render云平台免费部署ChatSydney

### DISCUSSION THREAD

**Discussion by renqabs** [viopsa233/AI-testers-Forum#68] [技术交流]
最近B站上看了一个render一键部署go-proxy-bingai的视频，我试了下搭建了一个ChatSydney，Render云平台提供了免费搭建应用的资源，512M RAM ，0.1 CPU，可运行时长每月750 hours，带宽每月200G，长时间不使用会休眠，有请求时自动唤醒，实测搭建ChatSydney还是没问题的，服务器是国外的，相当于自带魔法，有需要的小伙伴可以试试，有问题随时交流
1 github账号一个
2 登录[render](https://render.com/) （可使用github账号，也可使用邮箱注册）
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/97ce4c51-6749-49e0-b01d-48be45d28bf5)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/39700a87-08a2-4159-9ef1-6d42bd4f121c)

3 New -> Web Service
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/ac50c855-be4c-478a-aa22-843fdff1adac)

4 千万不要在右侧Configure account添加github账号（很重要，不然有封号风险），直接使用公开的代码仓库（直接使用我的也可以）
https://github.com/renqabs/ChatSydney-react-img.git
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/b52513e9-3794-44d4-8d01-3f6754e60669)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/811b7f60-1e0f-40ee-844d-e6a22ae5ab7d)

5 接下来，Name随便写英文字符就行，Region随便选一个（我一般选US的），Runtime选择docker
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/44060052-5b38-4479-bc24-e1efef5a6aeb)

6 点击Advanced，将Auto-Deploy改为No（防止因为我的代码变化导致你的docker自动重新部署）
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/69667ec8-8ef2-4de2-a1b5-a1710575645b)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/8237d078-6a7e-41b8-899a-0df9bd5b6ba0)

7 最后，点击Create Web Service会自动进行部署，等待部署完成后（大概3-5分钟），复制左上角的https链接，浏览器访问即可
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/82e0ec7d-3e09-431f-ba81-ced8866d3249)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/4fe4bf5d-e679-43ac-882a-f35898fb9b48)

**Comment by yinghua8wu** [comment]
为什么用docker部署会失败啊？

**Reply by renqabs** [reply]
> 为什么用docker部署会失败啊？

我也不清楚，我也是先用render上的docker Runtime部署的，但是部署的时候一直卡在那里，后来换成python就好了，可能是因为docker占用的资源比较多吧，不知道你的是什么情况，有没有具体的错误？

**Reply by yinghua8wu** [reply]
我知道原因了，docker默认部署在127.0.0.1上，必须部署在0.0.0.0上才行

**Comment by ufomonb** [comment]
哥，有详细教程吗？

**Reply by renqabs** [reply]
已更新

**Comment by synge0718** [comment]
试了一下replit，也部署成功了，并且replit没有时间限制，要用的时候开起来，不用的时候关掉就行了

**Reply by renqabs** [reply]
感谢分享，我也用replit部署成功了，还有一个平台也挺不错的，就是hugging face中space的docker,也是免费的，2 vCPU 16G RAM，可直接设置为private，也可以设置为public
![搜狗截图20230603203209](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/e06f1e95-e0ca-4562-b666-021e3f18d342)

![搜狗截图20230603202807](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/fc45d51b-f697-4b19-a768-c3cd725fb430)

**Comment by 2633381794** [comment]
第四步怎么修改代码呢？不会改的话就一直报错{"stack":"错误:构造'WebSocket'失败:不安全的WebSocket连接可能无法从通过HTTPS加载的页面发起。\n at <anonymous>:330:25\n at new Promise (<anonymous>)\n at <anonymous>:329:18\n at Generator。next (<anonymous>)\n at asyncGeneratorStep (<anonymous>:1:103)\n at _next (<anonymous>:2:194)\n at <anonymous>:2:364\n at new Promise (<anonymous>)\n用不了。

**Reply by renqabs** [reply]
需要下载InterestingDarkness大佬的ChatSydney代码，修改index.html文件后，再上传到自己的github仓库

**Comment by renqabs** [comment]
InterestingDarkness大佬的ChatSydney删库了，太可惜了。。。

**Reply by q2408808** [reply]
出什么事了吗？

**Reply by renqabs** [reply]
不清楚。。。

**Reply by q2408808** [reply]
他号都删了。贴吧好像也找不到他了 可怕。

**Reply by jonasreisner** [reply]
啊？发生什么事了

**Reply by gunshipwu** [reply]
收到了MS律师涵？ 按说从这的信息很难人肉到他吧。

**Comment by viopsa233** [comment]
> 收到了MS律师涵？ 按说从这的信息很难人肉到他吧。

不应该吧,隔壁的node-chatgpt-api都没事

**Comment by viopsa233** [comment]
> 他号都删了。贴吧好像也找不到他了 可怕。

[这个](https://tieba.baidu.com/home/main?id=tb.1.78446af.SlzESbzPE70h0itpBykWVg&fr=home)

**Reply by q2408808** [reply]
关注了也私信不了，很奇怪。而且我记得以前他的ID是正常ID吧 不是这种数字的

**Comment by viopsa233** [comment]
> 关注了也私信不了，很奇怪。而且我记得以前他的ID是正常ID吧 不是这种数字的

因为他注销了

**Reply by Cyrker** [reply]
为啥要这样嘞..

**Comment by rswwlhy** [comment]
你好我部署了，如果我想修改开头的“You're an AI assistant named Sydney. Instead of introducing yourself as Bing.....”。这段话去代码哪里修改？我目前在用作者你的，我打算复制到个人库里，从代码层面修改这部分的话，毕竟每次clear刷新后，又重置回来这句话了。实在是个新手。

**Reply by renqabs** [reply]
在public下index.html中修改，搜索一下就能定位到

**Comment by 2633381794** [comment]
【自动回复】抱歉，我现在有事不在，稍后再回邮件给您。
